### PR TITLE
fix(Callout): Calculate max height when the targetEdge is top using bottom and top bounds of the callout

### LINKED ISSUE #29319: [Bug]: Callout positioning does not account for scrollable content
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) ia32 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 20.71 GB / 31.95 GB
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (116.0.1938.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/nipope1/pen/jOXBOvO

## Actual Behavior
When `directionalHint = top...` and `directionalHintFixed = true`, and callout content exceeds the remaining height of the viewport, callout vertically fills the entire viewport.

(I put together this pen to confirm that the issue was specifically with callout, and not with dropdown: https://codepen.io/nipope1/pen/MWZGxzw)

## Expected Behavior
Callout should pin to the top of its target and fill the remaining space to the top of the viewport.

CodePen that shows working behavior in Fluent v7: https://codepen.io/nipope1/pen/poqVYaM

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

Azure Deployment Environments (Azure Portal extension)

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by sopranopillow**
|Previous|New|
|---|---|
|image|image|

## Related Issue(s)

- Fixes #29319

### CODE REVIEW

**Comment by sopranopillow** [comment]
Will bring this up offline, but there's some issues here. This is a partial fix that doesn't apply to every case. The maxHeight hook was introduced so that when the callout's content's height was reduced to the point there's no scrollbar the callout would resize. Specifically in these lines of code in CalloutContent.base.tsx:
```jsx
    const getBounds = useBounds(props, targetRef, targetWindow);
    const positions = usePositions(props, hostElement, calloutElement, targetRef, getBounds);
    const maxHeight = useMaxHeight(props, getBounds, positions);
    const [mouseDownOnPopup, mouseUpOnPopup] = useDismissHandlers(
      props,
      positions,
      hostElement,
      targetRef,
      targetWindow,
    );

    // do not set both top and bottom css props from positions
    // instead, use maxHeight
    const isForcedInBounds = positions?.elementPosition.top && positions?.elementPosition.bottom;
    const cssPositions = {
      ...positions?.elementPosition,
      maxHeight,
    };

    if (isForcedInBounds) {
      cssPositions.bottom = undefined;
    }
```
We could revert this change for the callout to be sized correctly, but losing the resize based on content is not worth the change. AFAIK this would be the best next thing. I am open to suggestions @smhigley, what are your thoughts?

With the provided information to the useMaxHeight hook, there's really not much you can calculate. The calculations aren't correct and they are estimates which open to issues as the one described in the bug. If we want to truly fix this bug and future proof it, we will need to find a better way to get the best of both worlds.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6384a9370b8c64501f3b376a0fb177bda84bcc16'>6384a9370b8c64501f3b376a0fb177bda84bcc16</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-596bdd63-b895-4c31-abbe-2923fcc96234.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-596bdd63-b895-4c31-abbe-2923fcc96234.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29650&commitId=ca304cfb500bd085e80d76526be69775ffd8a4ae&env=prod&clientType=FLUENTUI)


  <details><summary><b>Pivot - Overflow</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Root.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Root.Narrow - Overflow menu.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/BaseButton-mount_base.html">629</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/BaseButton-mount.html">614</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Breadcrumb-mount_base.html">1647</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Breadcrumb-mount.html">1688</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Checkbox-mount_base.html">1672</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Checkbox-mount.html">1770</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/CheckboxBase-mount_base.html">1522</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/CheckboxBase-mount.html">1483</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ChoiceGroup-mount_base.html">2918</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ChoiceGroup-mount.html">2973</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ComboBox-mount_base.html">666</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ComboBox-mount.html">671</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/CommandBar-mount_base.html">6359</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/CommandBar-mount.html">6339</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ContextualMenu-mount_base.html">12764</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ContextualMenu-mount.html">12866</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DefaultButton-mount_base.html">770</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DefaultButton-mount.html">767</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DetailsRow-mount_base.html">2128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DetailsRow-mount.html">2185</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DetailsRowFast-mount_base.html">2193</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DetailsRowFast-mount.html">2247</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DetailsRowNoStyles-mount_base.html">2012</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DetailsRowNoStyles-mount.html">2035</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Dialog-mount_base.html">2765</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Dialog-mount.html">2656</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DocumentCardTitle-mount_base.html">237</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/DocumentCardTitle-mount.html">219</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Dropdown-mount_base.html">2005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Dropdown-mount.html">2015</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/FocusTrapZone-mount_base.html">1136</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/FocusTrapZone-mount.html">1142</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/FocusZone-mount_base.html">1113</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/FocusZone-mount.html">1080</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedList-mount_base.html">42513</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedList-mount.html">42407</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedList-virtual-rerender_base.html">20286</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedList-virtual-rerender.html">20258</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">51418</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedList-virtual-rerender-with-unmount.html">51240</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedListV2-mount_base.html">222</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedListV2-mount.html">230</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedListV2-virtual-rerender_base.html">219</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedListV2-virtual-rerender.html">224</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">216</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">237</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/IconButton-mount_base.html">1100</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/IconButton-mount.html">1106</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Label-mount_base.html">344</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Label-mount.html">328</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Layer-mount_base.html">2760</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Layer-mount.html">2779</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Link-mount_base.html">402</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Link-mount.html">407</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/MenuButton-mount_base.html">926</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/MenuButton-mount.html">928</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/MessageBar-mount_base.html">21896</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/MessageBar-mount.html">21908</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Nav-mount_base.html">1999</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Nav-mount.html">1942</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/OverflowSet-mount_base.html">781</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/OverflowSet-mount.html">780</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Panel-mount_base.html">1769</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Panel-mount.html">1771</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Persona-mount_base.html">728</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Persona-mount.html">764</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Pivot-mount_base.html">892</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Pivot-mount.html">887</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/PrimaryButton-mount_base.html">831</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/PrimaryButton-mount.html">842</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Rating-mount_base.html">4576</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Rating-mount.html">4535</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SearchBox-mount_base.html">927</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SearchBox-mount.html">899</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Shimmer-mount_base.html">1865</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Shimmer-mount.html">1871</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Slider-mount_base.html">1291</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Slider-mount.html">1313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SpinButton-mount_base.html">2911</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SpinButton-mount.html">2904</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Spinner-mount_base.html">393</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Spinner-mount.html">395</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SplitButton-mount_base.html">1819</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SplitButton-mount.html">1815</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Stack-mount_base.html">426</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Stack-mount.html">421</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/StackWithIntrinsicChildren-mount_base.html">845</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/StackWithIntrinsicChildren-mount.html">877</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/StackWithTextChildren-mount_base.html">2644</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/StackWithTextChildren-mount.html">2613</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SwatchColorPicker-mount_base.html">6124</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/SwatchColorPicker-mount.html">6251</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/TagPicker-mount_base.html">1448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/TagPicker-mount.html">1449</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Text-mount_base.html">382</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Text-mount.html">380</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/TextField-mount_base.html">935</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/TextField-mount.html">956</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ThemeProvider-mount_base.html">846</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ThemeProvider-mount.html">859</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ThemeProvider-virtual-rerender_base.html">580</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ThemeProvider-virtual-rerender.html">585</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1297</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Toggle-mount_base.html">621</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/Toggle-mount.html">622</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/buttonNative-mount_base.html">189</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29650/perf-test/buttonNative-mount.html">199</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Comment by spmonahan** [comment]
Hi @nipope, I know you've spent some time looking at this code. Would appreciate any input you have!

File: packages/react/src/components/Callout/CalloutContent.base.tsx [hunk 2]
```diff
@@ -126,19 +126,18 @@ function useMaxHeight(
     const { top: topBounds, bottom: bottomBounds } = getBounds() ?? {};
     let calculatedHeight: number | undefined;
 
-    if (typeof top === 'number' && bottomBounds) {
+    if (positions?.targetEdge === RectangleEdge.top && bottom && top && topBounds) {
```
Comment (sopranopillow) [inline packages/react/src/components/Callout/CalloutContent.base.tsx:129]:
```suggestion
    if (positions?.targetEdge === RectangleEdge.top && typeof bottom  === 'number' && typeof top  === 'number'  && topBounds) {
```

**Comment by sopranopillow** [comment]
After talking offline, we came up with a possible solution to the resize based on content and this bug. Instead of using the useMaxHeight hook, what if we internally add a resize observer to calculate the resizing and triggering the rerender? This of course needs to be a clear improvement and no regression. If there's any regression, we need to follow this fix instead.

**Comment by nipope** [comment]
@sopranopillow This seems like a good idea! Because there are already maxHeight calculations happening in `positioning.ts`, triggering a rerender on content resize would leverage that maxHeight logic.

**Pull request closed without merging**